### ISSUE TITLE
IPv4/IPv6 "data type domains" are barely usable

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#27491]
```
:) SELECT CAST('1.1.1.1' AS IPv4)

SELECT CAST('1.1.1.1', 'IPv4')

Query id: 1cebacff-513b-4f33-9b2f-57bd37b56e71


0 rows in set. Elapsed: 0.002 sec. 

Received exception from server (version 21.9.1):
Code: 6. DB::Exception: Received from localhost:9000. DB::Exception: Cannot parse string '1.1.1.1' as IPv4: syntax error at position 1 (parsed just '1'). Note: there are toIPv4OrZero and toIPv4OrNull functions, which returns zero/NULL instead of throwing exception.: While processing CAST('1.1.1.1', 'IPv4'). (CANNOT_PARSE_TEXT)
```

**Comment by alexey-milovidov** [comment]
Tech debt after #3669.

**Comment by novikd** [comment]
Fixed in #29224
